src/components/notification-tray/notification-tray.tsx

```diff
@@ -0,0 +1,82 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Listen,
+  Prop,
+} from "@stencil/core"
+
+@Component({
+  tag: "oui-notification-tray",
+  styleUrl: "notification-tray.css",
+})
+export class NotificationTray {
+  @Element() public tray: HTMLElement
+
+  /**
+   * Open or close the notification drawer
+   */
+  @Prop({ reflect: true, mutable: true }) public opened = false
+
+  /**
+   * Direction of the drawer
+   */
+  @Prop() public direction: "to-right" | "to-left" = "to-left"
+
+  /**
+   *
+   */
+  @Prop({ reflect: true, mutable: true }) public count: number = 0
+
+  /**
+   *
+   */
+  @Prop({ reflect: true, mutable: true }) public unread: number = 0
+
+  /**
+   *
+   */
+  @Event() public dismissall: EventEmitter
+  public dismissallHandler = () => {
+    this.dismissall.emit()
+  }
+
+  /**
+   *
+   */
+  @Event() public dismiss: EventEmitter
```

Comment:
```suggestion
  @Event() public dismiss: EventEmitter<string>
```
This generic type can be better expressed by specifying "string" as the type parameter (custom element detail type), which tells the subscribing event handlers in typescript what type the event detail is, and can be used to validate the handler code

Reply:
done :)